File: modules/spring-core/testdata/java/service/packageScanRunConfigurationAppConfigConstructor/com/app/MainClass.java
package com.app;

import com.outer.AppTestConfiguration;
import com.outer.OuterComponent;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;

public class MainClass {
    public static void main(String[] args) {
        try (var context = new AnnotationConfigApplicationContext(AppTestConfiguration.class)) {
            var bean = context.getBean(OuterComponent.class);
        }
    }
}

File: modules/spring-core/testdata/kotlin/service/packageScanRunConfigurationAppConfigConstructor/com/app/MainClass.kt
package com.app;

import com.outer.AppTestConfiguration;
import com.outer.OuterComponent;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;

class MainClass

fun main() {
    AnnotationConfigApplicationContext(AppTestConfiguration::class)
        .use { context ->
            val bean = context.getBean(OuterComponent::class.java)
        }
}
